src/v0/destinations/fb_custom_audience/transformV2.js

```diff
@@ -0,0 +1,312 @@
+
+  if (isRaw) {
+    paramsPayload.is_raw = isRaw;
+  }
+
+  const dataSource = getDataSource(type, subType);
+  if (Object.keys(dataSource).length > 0) {
+    paramsPayload.data_source = dataSource;
+  }
+
+  const groupedRecordsByAction = lodash.groupBy(groupedRecordInputs, (record) =>
+    record.message.action?.toLowerCase(),
+  );
+
+  const finalResponse = [];
+
+  let insertResponse;
+  let deleteResponse;
+  let updateResponse;
+
+  if (groupedRecordsByAction.delete) {
+    const deleteRecordChunksArray = returnArrayOfSubarrays(
+      groupedRecordsByAction.delete,
+      maxUserCountNumber,
+    );
+    deleteResponse = processRecordEventArray(
+      deleteRecordChunksArray,
+      userSchema,
+      isHashRequired,
+      disableFormat,
+      paramsPayload,
+      prepareParams,
+      destination,
+      'remove',
+      operationAudienceId,
+    );
+  }
+
+  if (groupedRecordsByAction.insert) {
+    const insertRecordChunksArray = returnArrayOfSubarrays(
+      groupedRecordsByAction.insert,
+      maxUserCountNumber,
+    );
+
+    insertResponse = processRecordEventArray(
+      insertRecordChunksArray,
+      userSchema,
+      isHashRequired,
+      disableFormat,
+      paramsPayload,
+      prepareParams,
+      destination,
+      'add',
+      operationAudienceId,
+    );
+  }
+
+  if (groupedRecordsByAction.update) {
+    const updateRecordChunksArray = returnArrayOfSubarrays(
+      groupedRecordsByAction.update,
+      maxUserCountNumber,
+    );
+    updateResponse = processRecordEventArray(
+      updateRecordChunksArray,
+      userSchema,
+      isHashRequired,
+      disableFormat,
+      paramsPayload,
+      prepareParams,
+      destination,
+      'add',
+      operationAudienceId,
+    );
+  }
+
+  const eventTypes = ['update', 'insert', 'delete'];
+  const errorMetaData = [];
+  const errorMetaDataObject = getErrorMetaData(groupedRecordsByAction, eventTypes);
+  if (errorMetaDataObject.length > 0) {
+    errorMetaData.push(errorMetaDataObject);
+  }
+
+  const error = new InstrumentationError('Invalid action type in record event');
+  const errorObj = generateErrorObject(error);
+  const errorResponseList = errorMetaData.map((metadata) =>
+    getErrorRespEvents(metadata, errorObj.status, errorObj.message, errorObj.statTags),
+  );
+
+  if (deleteResponse && deleteResponse.batchedRequest.length > 0) {
+    finalResponse.push(deleteResponse);
+  }
+  if (insertResponse && insertResponse.batchedRequest.length > 0) {
+    finalResponse.push(insertResponse);
+  }
+  if (updateResponse && updateResponse.batchedRequest.length > 0) {
+    finalResponse.push(updateResponse);
+  }
+  if (errorResponseList.length > 0) {
+    finalResponse.push(...errorResponseList);
+  }
+
+  if (finalResponse.length === 0) {
+    throw new InstrumentationError(
+      'Missing valid parameters, unable to generate transformed payload',
+    );
+  }
+  return finalResponse;
+}
+
+module.exports = {
+  processRecordInputs,
```

Comment:
Move responseBuilderSimple & getDataSource to util class as they are common in both audienceList & recordType flow
Rename `transformV2.js` to `recordTransform.js` as v2 class is not a complete parallel path to transform.js

Reply:
addressed